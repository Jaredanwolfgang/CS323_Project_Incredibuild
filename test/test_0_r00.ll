; ModuleID = 'test_0_r00.spl'
source_filename = "test_0_r00.spl"

define i32 @main() {
entry:
  %a = alloca i32, align 4
  store i32 3, ptr %a, align 4
  br label %cond

cond:                                             ; preds = %merge3, %entry
  br i1 true, label %body, label %merge

body:                                             ; preds = %cond
  %a1 = load i32, ptr %a, align 4
  %addtmp = add i32 %a1, 1
  store i32 %addtmp, ptr %a, align 4
  %a2 = load i32, ptr %a, align 4
  %eqtmp = icmp eq i32 %a2, 5
  br i1 %eqtmp, label %then, label %merge3

merge:                                            ; preds = %then, %cond
  %a4 = load i32, ptr %a, align 4
  ret i32 %a4

then:                                             ; preds = %body
  br label %merge

merge3:                                           ; preds = %body
  br label %cond
}
