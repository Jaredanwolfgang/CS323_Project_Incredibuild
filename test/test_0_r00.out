Function: main:[Body: [Variable Declaration: a = [0: u32] with dimensions []; Variable Assignment: a = 3: u32, While Loop (Condition: true):
do Body: [Variable Assignment: a = (a + 1: u32), If: Condition: a == 5: u32 then Body: [Break]], Return: a]]
